src/display/annotation_layer.js

```diff
@@ -198,6 +199,92 @@ class AnnotationElement {
     return AnnotationElement._hasPopupData(this.data);
   }
 
+  get hasCommentButton() {
+    return this.enableComment && this._isEditable && this.hasPopupElement;
+  }
+
+  get commentButtonPosition() {
+    const { quadPoints, rect } = this.data;
+    let maxX = -Infinity;
+    let maxY = -Infinity;
+    if (quadPoints?.length >= 4) {
+      for (let i = 0; i < quadPoints.length; i += 8) {
+        if (quadPoints[i + 1] > maxY) {
+          maxY = quadPoints[i + 1];
+          maxX = quadPoints[i + 2];
+        } else if (quadPoints[i + 1] === maxY) {
+          maxX = Math.max(maxX, quadPoints[i + 2]);
+        }
+      }
+      return [maxX, maxY];
+    }
+    if (rect) {
+      return [rect[2], rect[3]];
+    }
+    return null;
+  }
+
+  get commentButtonColor() {
+    if (!this.data.color) {
+      return null;
+    }
+    const [r, g, b] = this.data.color;
+    const opacity = this.data.opacity ?? 1;
+    const oppositeOpacity = 255 * (1 - opacity);
+
+    return this.changeLightness(
+      Math.min(r + oppositeOpacity, 255),
+      Math.min(g + oppositeOpacity, 255),
+      Math.min(b + oppositeOpacity, 255)
+    );
+  }
+
+  changeLightness(r, g, b) {
+    r /= 255;
+    g /= 255;
+    b /= 255;
+
+    const max = Math.max(r, g, b);
+    const min = Math.min(r, g, b);
+    const l = (max + min) / 2;
+    const newL = (((1 + Math.sqrt(l)) / 2) * 100).toFixed(2);
+
+    if (max === min) {
+      // gray
+      return `hsl(0, 0%, ${newL}%)`;
+    }
+
+    const d = max - min;
+
+    // hue (branch on max only; avoids mod)
+    let h;
+    if (max === r) {
+      h = (g - b) / d + (g < b ? 6 : 0);
+    } else if (max === g) {
+      h = (b - r) / d + 2;
+    } else {
+      // max === b
+      h = (r - g) / d + 4;
+    }
+    h = (h * 60).toFixed(2);
+    const s = ((d / (1 - Math.abs(2 * l - 1))) * 100).toFixed(2);
+
+    return `hsl(${h}, ${s}%, ${newL}%)`;
+  }
+
+  normalizePoint(point) {
```

Comment:
This one can also be made private.

Reply:
Unfortunately, no we can't but we can at least add an underscore.